src/Compilers/CSharp/Portable/BoundTree/BoundDecisionDag.cs

```diff
@@ -84,7 +84,7 @@ public ImmutableArray<BoundDecisionDagNode> TopologicallySortedNodes
         /// takes as its input the node to be rewritten and a function that returns the previously computed
         /// rewritten node for successor nodes.
         /// </summary>
-        public BoundDecisionDag Rewrite(Func<BoundDecisionDagNode, Func<BoundDecisionDagNode, BoundDecisionDagNode>, BoundDecisionDagNode> makeReplacement)
+        public BoundDecisionDag Rewrite(Func<BoundDecisionDagNode, IReadOnlyDictionary<BoundDecisionDagNode, BoundDecisionDagNode>, BoundDecisionDagNode> makeReplacement)
```

Comment:
>IReadOnlyDictionary [](http://example.com/codeflow?start=67&length=19)

Is this refactoring still necessary for the purpose of this PR? If not, please revert or extract to a different PR.